MdeModulePkg/Core/Pei/Dispatcher/Dispatcher.c

```diff
@@ -1411,6 +1413,24 @@ PeiDispatcher (
   PeimFileHandle = NULL;
   EntryPoint     = 0;
 
+  if (NULL == Private->DelayedDispatchTable) {
+    // MSCHANGE Start
+    GuidHob = GetFirstGuidHob (&gEfiDelayedDispatchTableGuid);
+    if (NULL != GuidHob) {
+      Private->DelayedDispatchTable = (DELAYED_DISPATCH_TABLE *)(GET_GUID_HOB_DATA (GuidHob));
+    } else {
+      UINT32  TableSize = sizeof (DELAYED_DISPATCH_TABLE) + ((FixedPcdGet32 (PcdDelayedDispatchMaxEntries) - 1) * sizeof (DELAYED_DISPATCH_ENTRY));
+      Private->DelayedDispatchTable = BuildGuidHob (&gEfiDelayedDispatchTableGuid, TableSize);
+      if (NULL != Private->DelayedDispatchTable) {
+        ZeroMem (Private->DelayedDispatchTable, TableSize);
+        Status = PeiServicesInstallPpi (&mDelayedDispatchDesc);
+        ASSERT_EFI_ERROR (Status);
+        Status = PeiServicesNotifyPpi (&mDelayedDispatchNotifyDesc);
+        ASSERT_EFI_ERROR (Status);
+      }
+    }                                                                    // MS_CHANGE End

```

Comment:
While you are making changes, can you change the MS_CHANGEs to MU_CHANGEs? Also, the end tag should go on the next brace down.